### ISSUE TITLE
Inconsistent PPO entropy and KL with ray 0.7.6 and 0.8.0

### ISSUE THREAD

**Issue by jinbo-huang** [ray-project/ray#7399]
### Inconsistent PPO entropy and KL with ray 0.7.6 and 0.8.0

Ray version: 0.7.6 and 0.8.0
Python version: 3.7
OS: Ubuntu 18.04

Hi all,
I'm training ppo agents in a discrete action space environment and I found that with exactly the same ppo training configurations, the updates seemed quite different between version 0.7.6 and 0.8.0. As you can see in the images below, the 1st image shows that the blue one (ray 0.7.6) has higher kl than the orange one (ray 0.8.0). In the 2nd image, the entropy of the blue one (ray 0.7.6) decreases much faster than the orange one (ray 0.8.0)

![image](https://user-images.githubusercontent.com/4936182/75669318-cf997c80-5cb5-11ea-9d80-2e9010043f93.png)
![image](https://user-images.githubusercontent.com/4936182/75669567-30c15000-5cb6-11ea-90c8-6f96939dabe9.png)

Here is part of my configurations:
```
lr: 0.0005
gamma: 0.99
entropy_coeff: 0.01
preprocessor_pref: rllib
clip_param: 0.2
sgd_minibatch_size: 512
train_batch_size: 8192
sample_batch_size: 200
num_sgd_iter: 10
lambda: 0.95
kl_coeff: 0.0
lr_schedule: [[0, 0.0001], [500000, 0.00001]]
entropy_coeff_schedule: 500000
```

I know the training would not always be the same in RL, but I tried 3~5 times and they all showed the same problems.
I've checked the ppo configurations from 0.7.6 and 0.8.0 and they looked the same. Did I miss any new ppo configs?

**Comment by ericl** [comment]
Maybe it's https://github.com/ray-project/ray/pull/6475 ?

Try using the latest version 0.8.2, and see if it affects any standard benchmarks.

**Comment by iceman126** [comment]
Thanks for your advice. I will train another agent with version 0.8.2. 

About #6475, the environment I use is discrete action space (`Categorical` class in rllib), would this affect the entropy value of `Categorical` class?

**Comment by iceman126** [comment]
I found the reason. Because of commit 1f043daf697517e85e6885bfbed4037990e0ec44 fixed the bug of using the incorrect optimizer, the learning started decaying, that's the entropy didn't decrease. 

BTW, before the bug was fixed, the optimizer always used the learning rate at `lr`? It never decays?
